File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> dots) {
    int answer = 0;
    vector<int> x;
    vector<int> y;
    
    for(auto dot:dots)
    {
        x.push_back(dot[0]);
        y.push_back(dot[1]);
    }
    
    sort(x.begin(),x.end());
    sort(y.begin(),y.end());
    
    int width=(x[3]>x[0])?x[3]-x[0]:x[0]-x[3];
    int height=(y[3]>y[0])?y[3]-y[0]:y[0]-y[3];
    
    answer=width*height;
    
    return answer;
}

File: 프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.py
def solution(dots):
    answer = 0
    x=[]
    y=[]
    
    for dot in dots:
        x.append(dot[0])
        y.append(dot[1])
        
    answer=(max(x)-min(x))*(max(y)-min(y))
        
    
    return answer